### PR TITLE
[OCPCLOUD-1089] Simple e2e test suite for CCCMO clusterOperator 

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Make sure the `ClusterOperator` is provisioned, have necessary fields populated and marked available even when there is no infrastructure in place to provision.

### CODE REVIEW

File: controllers/clusteroperator_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+	type testCase struct {
+		releaseVersionEnvVariableValue string
+		namespace                      string
+		existingCO                     *configv1.ClusterOperator
+	}
+
+	DescribeTable("should ensure Cluster Operator status is present",
+		func(tc testCase) {
+			By("Running test case")
```
Comment (JoelSpeed) [inline controllers/clusteroperator_controller_test.go:55]: By statements are more useful when they describe parts of the test, so for example you can do:
```
By("Setting the release version", func() {
  expectedVersion := unknownVersionValue
  if tc.releaseVersionEnvVariableValue != "" {
    expectedVersion = tc.releaseVersionEnvVariableValue
    Expect(os.Setenv(releaseVersionEnvVariableName, tc.releaseVersionEnvVariableValue)).To(Succeed())
  }
})
```
Currently this By doesn't add much value

File: controllers/clusteroperator_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+				Expect(err).To(Succeed())
+			}
+
+			_, err := operatorController.Reconcile(context.Background(), reconcile.Request{})
+			Expect(err).To(Succeed())
+
+			getOp := &configv1.ClusterOperator{}
+			Eventually(func() (bool, error) {
+				var err error
+				err = cl.Get(context.Background(), client.ObjectKey{Name: clusterOperatorName}, getOp)
```
Comment (JoelSpeed) [inline controllers/clusteroperator_controller_test.go:77]: Can inline this
```suggestion
				err := cl.Get(context.Background(), client.ObjectKey{Name: clusterOperatorName}, getOp)
```

File: controllers/clusteroperator_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+		operator.SetName(clusterOperatorName)
+	})
+
+	AfterEach(func() {
+		os.Unsetenv(releaseVersionEnvVariableName)
+		err := cl.Get(context.Background(), client.ObjectKey{Name: clusterOperatorName}, &configv1.ClusterOperator{})
+		if err == nil || !apierrors.IsNotFound(err) {
+			Expect(cl.Delete(context.Background(), operator)).To(Succeed())
+		}
+		Expect(apierrors.IsNotFound(cl.Get(context.Background(), client.ObjectKey{Name: clusterOperatorName}, &configv1.ClusterOperator{}))).To(BeTrue())
```
Comment (JoelSpeed) [inline controllers/clusteroperator_controller_test.go:44]: I would suggest using async calls here to make sure these are reliable

```
Eventually(func() error {
  err := cl.Delete(context.Background(), operator)
  if err != nil && !apierrors.IsNotFound(err) {
    return err
  }
  return nil
}).Should(Succeed())
Eventually(func() error {
  err := cl.Get(context.Background(), client.ObjectKey{Name: clusterOperatorName}, &configv1.ClusterOperator{})
  if apierrors.IsNotFound(err) {
    return nil
  }
  return errors.New("object still exists")
}).Should(Succeed())
```

**Review by Fedosin — APPROVED** [review]
/lgtm
/approve

**Pull request merged**